database.py

```diff
@@ -561,6 +561,18 @@ def mark_submission_invalid(self, submission_id: SubmissionId, /) -> None:
 
             self.refresh_user_best_runs(year, day, part, user)
 
+    def pick_new_container_versions(self, versions: list[str]) -> set[str]:
+        """
+        Given a list of container versions, return those not in the DB already.
+        """
+        query = "SELECT container_version FROM container_versions"
+        given_versions = set(versions)
```

Comment:
why are we asking for a list here and then immediately converting to a set? we should just take a set directly